### PR TITLE
Secondary boot handling

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

File: core/arch/arm/plat-imx/imx_sip_caller.c [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		  "+m" (args->regs[2]), "+m" (args->regs[3])
+		: "m" (args->regs[4]), "m" (args->regs[5]),
+		  "m" (args->regs[6])
+		: "x0", "x1", "x2", "x3", "x4", "x5", "x6", "x7",
+		  "x8", "x9", "x10", "x11", "x12", "x13", "x14", "x15",
+		  "x16", "x17");
+}
+
+
+unsigned long call_imx_sip(unsigned long id, unsigned long reg0,
```
Comment (igoropaniuk) [inline core/arch/arm/plat-imx/imx_sip_caller.c:54]: that's a copy-paste of SMC call wrappers from U-Boot , probably there should be something similar in OP-TEE, however I didn't find it

Comment (ricardosalveti) [reply core/arch/arm/plat-imx/imx_sip_caller.c:54]: U-boot has a difference license, so this code would need to be based on something that is compatible with the license used by optee-os. Should be fine if you find similar wrappers somewhere else.

**Comment by ricardosalveti** [comment]
Should we have it as a PTA? I think it makes sense, but at the same time I know we're adding fiovb as an early TA as we also need to use the normal TA on u-boot.

`CFG_IN_TREE_EARLY_TAS=fiovb/22250a54-0bf1-48fe-8002-7b20f1c9c9b1`

File: core/arch/arm/plat-imx/imx_sip_caller.c (hunk 1 above)
Comment (igoropaniuk) [reply core/arch/arm/plat-imx/imx_sip_caller.c:54]: I've added initially jus to check if it even works at all. Yes, will do

**Comment by igoropaniuk** [comment]
we need PTA to make calls to TF-A (as PTA is just linked with optee-os core), which is not possible to do from a regular TA. So communication is:
TA->PTA->TF-A

File: core/pta/fiovb.c [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+static TEE_Result invoke_command(void *pSessionContext __unused,
+				 uint32_t nCommandID, uint32_t pt,
+				 TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INOUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	uint32_t secondary;
```
Comment (ricardosalveti) [inline core/pta/fiovb.c:24]: Won't this cause a defined but not used warning when not building for mx8mm/mx8mq?

Also good to initialize it as well.

**Review by ricardosalveti — APPROVED** [review]

**Pull request merged**